### DISCUSSION TITLE
Cal command

### DISCUSSION THREAD

**Discussion by debajyoti1990** [nushell/nushell#8787] [Ideas]
Hi there, 

Is there any plan for making cal command have features like linux's cal ? Aeking as I have windows in work computer and cygwin is slightly overkill for only calender application.

**Comment by fdncred** [comment]
What Linux features are you referring to?

**Reply by debajyoti1990** [reply]
Week number view(very much useful) and current day highlight(not that much important but good to have). In Linux, my usual command in bash is 'cal -w -m -y' , I alias 'cal -m -w' to cw and so it just becomes 'cw -y'

**Comment by ghost** [comment]
A feature comparison:

util-linux cal | nushell cal
--- | ---
`-1`, `-3` `-n MONTH-COUNT` | missing (`--full-year`, but thats often overkill)
`--span`: display months spanning the date | missing (`--full-year 20xx`, but thats a whole year)
`--sunday`, `--monday` | `--week-start`
`--vertical` | `--rotate` i guess
`--iso`, `--julian`, `--reform` | missing (but probably not needed often)
`--year` | `--full-year`
`--twelve`: next 12 months | missing
`--week`: week numbers | missing (it numbers them, but except for january wrong)
`cal 2023` | `cal --full-year 2023`
`cal tomorrow`, `cal august`, `cal 11 2023`, `cal 13 11 2023` | missing

**Reply by fdncred** [reply]
I don't see nushell implementing all these things. If someone wanted to implement a new flag, that may be ok but we don't really want to implement 50 flags on every built-in like most bash utilities. We like to be more composable via our pipeline infrastructure.

**Comment by mykhal** [comment]
`cal` does not seem to be part of POSIX, but common usual args pattern on UN*X/*BSD/Linux systems seem to be

```
cal [[month] year]
```

e.g.

```
$ cal 5 2019
      May 2019
Su Mo Tu We Th Fr Sa
          1  2  3  4
 5  6  7  8  9 10 11
12 13 14 15 16 17 18
19 20 21 22 23 24 25
26 27 28 29 30 31   
```

So maybe the help or the book could mention that this can be emulated consistently to internal nu cal behavior with
```
cal -m --as-table --full-year 2019 | where month == 5 | reject month | table -i false
```
(or some simpler way, if exists)

(Current Friday 13th's [example](https://github.com/nushell/nushell/blob/bf23573fa93f5426c1de8a6f843e03dad2e6a57c/crates/nu-command/src/generators/cal.rs#L94) there is nice and somewhat orthogonal, as is uses `where` on day of week instead. It's a keeper.)
